### DISCUSSION TITLE
How to delete a map from an array

### DISCUSSION THREAD

**Discussion by carlosatrios** [mikefarah/yq#1679] [Q&A]
**Describe the bug**

Can't remove an element from a yaml. 

**Input Yaml**
```
channels: 
  - pytorch 
  - nvidia
  - intel
  - robostack
  - defaults
  - conda-forge

dependencies:
    - cmake
    - python=3.8.16
    - numpy
    - vcstool
    - pybind11    
    - pip:
      - omegaconf
      - boto3  

prefix: /opt/miniconda
``` 

**Command**
The command you ran:
```
yq eval '.dependencies' env.yml -o y | yq eval 'del(.pip)'
```

**Actual behavior**

```
Error: cannot index array with 'pip' (strconv.ParseInt: parsing "pip": invalid syntax)
```

**Expected behavior**
```
    - cmake
    - python=3.8.16
    - numpy
    - vcstool
    - pybind11    
``` 

**Additional context**

**Comment by mikefarah** [comment]
TLDR:

```
yq '.dependencies | del(.[] | select(has("pip")))' file.yaml 
```

Explanation:
`dependencies` is an array, where each entry is a string _except_ the entry with `pip` in it - that's actually an object with a single key entry "pip".

- `.dependencies` traverses into that array
- The `del` function works by giving it the path to the nodes to delete as an argument. So we need to find the entry that is a map that has a key "pip"
- `.[]` matches all elements of dependencies
- `select(has("pip"))` will filter all all entries, except the one with pip in it
